src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2

```diff
@@ -44,27 +45,40 @@ def start():
             render_message("user", kwargs)
         # Append chat history to kwargs
         response = run_flow({'{{chat_history_input_name}}': get_chat_history_from_session(), **kwargs})
+
         # Get base64 for multi modal object
-        resolved_outputs = invoker._convert_multimedia_data_to_base64(response)
+        resolved_outputs = {
+            k: convert_multimedia_data_to_base64(v, with_type=True, dict_type=True)
+            for k, v in response.items()
+        }
         st.session_state.messages.append(("assistant", resolved_outputs))
-        session_state_history.update({"outputs": response.output})
+        session_state_history.update({"outputs": response})
         st.session_state.history.append(session_state_history)
-        invoker._dump_invoke_result(response)
+
+        dump_path = Path('{{flow_path}}').parent
+        {% if is_chat_flow %}
+        dump_file_prefix = "chat"
+        {% else %}
+        dump_file_prefix = "flow"
+        {% endif %}
+        response = persist_multimedia_data(
+            response, base_dir=dump_path, sub_dir=Path(".promptflow/output")
+        )
+
+        # need discussion on how to dump flow result
```

Comment:
looks like we should create a new api which returns line result or should just dump the flow result